pkg/gardenlet/operation/botanist/apiserverproxy.go

```diff
@@ -54,5 +56,15 @@ func (b *Botanist) DeployAPIServerProxy(ctx context.Context) error {
 
 	b.Shoot.Components.SystemComponents.APIServerProxy.SetAdvertiseIPAddress(b.APIServerClusterIP)
 
-	return b.Shoot.Components.SystemComponents.APIServerProxy.Deploy(ctx)
+	if err := b.Shoot.Components.SystemComponents.APIServerProxy.Deploy(ctx); err != nil {
+		return err
+	}
+
+	return b.Shoot.UpdateInfoStatus(ctx, b.GardenClient, true, func(shoot *gardencorev1beta1.Shoot) error {
+		c := clock.RealClock{}
```

Comment:
Can you attach a clock to the `Botanist` type?
It should be set in the `botanist.New` function. I only see one occurrence of a clock in the `botanist` package in the `New` function which can be replaced by the new `Botanist.Clock` member.